tgui/packages/tgui/interfaces/CameraConsole.tsx

```diff
@@ -0,0 +1,199 @@
+import { filter, sortBy } from 'common/collections';
+import { flow } from 'common/fp';
+import { BooleanLike, classes } from 'common/react';
+import { createSearch } from 'common/string';
+import { useBackend, useLocalState } from '../backend';
+import { Button, ByondUi, Input, NoticeBox, Section, Stack } from '../components';
+import { Window } from '../layouts';
+
+type Data = {
+  can_spy: BooleanLike;
+  mapRef: string;
+  cameras: Camera[];
+  activeCamera: Camera & { status: BooleanLike };
+  network: string[];
+};
+
+type Camera = {
+  name: string;
+};
+
+/**
+ * Returns previous and next camera names relative to the currently
+ * active camera.
+ */
+const prevNextCamera = (
+  cameras: Camera[],
+  activeCamera: Camera & { status: BooleanLike }
+) => {
+  if (!activeCamera) {
+    return [];
+  }
+  const index = cameras.findIndex(
+    (camera) => camera?.name === activeCamera.name
+  );
+  return [cameras[index - 1]?.name, cameras[index + 1]?.name];
+};
+
+/**
+ * Camera selector.
+ *
+ * Filters cameras, applies search terms and sorts the alphabetically.
+ */
+const selectCameras = (cameras: Camera[], searchText = ''): Camera[] => {
+  const testSearch = createSearch(searchText, (camera: Camera) => camera.name);
+
+  return flow([
+    // Null camera filter
+    filter((camera: Camera) => !!camera?.name),
```

Comment:
it's possible that the issue was here. without digging into the backend and just reworking how this works entirely (null camera names causes another issue), typescript was complaining that this needed to be casted into a boolean